spec/services/vine_voucher_validator_service_spec.rb

```diff
@@ -0,0 +1,365 @@
+      let!(:vine_connected_app) {
+        ConnectedApps::Vine.create(
+          enterprise: distributor, data: { api_key: "1234568", secret: "my_secret" }
+        )
+      }
+      let(:data) {
+        {
+          meta: { responseCode: 200, limit: 50, offset: 0, message: "" },
+          data: {
+            id: "9d2437c8-4559-4dda-802e-8d9c642a0c1d",
+            voucher_short_code: voucher_code,
+            voucher_set_id: "9d24349c-1fe8-4090-988b-d7355ed32559",
+            is_test: 1,
+            voucher_value_original: 500,
+            voucher_value_remaining: '',
+            num_voucher_redemptions: 0,
+            last_redemption_at: "null",
+            created_at: "2024-10-01T13:20:02.000000Z",
+            updated_at: "2024-10-01T13:20:02.000000Z",
+            deleted_at: "null"
+          }
+        }.deep_stringify_keys
+      }
+
+      before do
+        allow(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: true, status: 200, data: )
+        )
+      end
+
+      it "returns an invalid voucher" do
+        voucher = validate_voucher_service.validate
+        expect(voucher).not_to be_valid
+      end
+    end
+
+    context "with an existing voucher" do
+      let!(:vine_connected_app) {
+        ConnectedApps::Vine.create(
+          enterprise: distributor, data: { api_key: "1234567", secret: "my_secret" }
+        )
+      }
+      let!(:voucher) {
+        create(:voucher_flat_rate, enterprise: distributor, code: voucher_code,
+                                   amount: 500, voucher_type: "VINE" )
+      }
+
+      let(:data) {
+        {
+          meta: { responseCode: 200, limit: 50, offset: 0, message: "" },
+          data: {
+            id: "9d2437c8-4559-4dda-802e-8d9c642a0c1d",
+            voucher_short_code: voucher_code,
+            voucher_set_id: "9d24349c-1fe8-4090-988b-d7355ed32559",
+            is_test: 1,
+            voucher_value_original: 500,
+            voucher_value_remaining: 250,
+            num_voucher_redemptions: 1,
+            last_redemption_at: "2024-10-05T13:20:02.000000Z",
+            created_at: "2024-10-01T13:20:02.000000Z",
+            updated_at: "2024-10-01T13:20:02.000000Z",
+            deleted_at: "null"
+          }
+        }.deep_stringify_keys
+      }
+
+      before do
+        allow(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: true, status: 200, data: )
+        )
+      end
+
+      it "verify the voucher with VINE API" do
+        expect(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: true, status: 200, data: )
+        )
+
+        validate_voucher_service.validate
+      end
+
+      it "updates the VINE voucher" do
+        vine_voucher = validate_voucher_service.validate
+
+        expect(vine_voucher.id).to eq(voucher.id)
+        expect(vine_voucher.amount).to eq(2.50)
```

Comment:
We probably want to know that the value was saved to the DB too, so I suggest a `reload` here.